hgtv/views/playlist.py

```diff
@@ -181,20 +182,33 @@ def playlist_edit(channel, playlist):
     (Playlist, {'name': 'playlist', 'channel': 'channel'}, 'playlist'),
     permission='delete'
     )
+@render_with({'text/html': 'index.html.jinja2'}, json=True)
 def playlist_delete(channel, playlist):
-    return render_delete_sqla(playlist, db, title=u"Confirm delete",
-        message=u"Delete playlist '%s'? This cannot be undone." % playlist.title,
-        success=u"You have deleted playlist '%s'." % playlist.title,
-        next=channel.url_for())
+    if request.method == 'GET':
+        return {'playlist': dict(playlist.current_access())}
+    form = Form()
+    if form.validate_on_submit():
+        db.session.delete(playlist)
```

Comment:
This will delete all videos in the playlist, but won't remove stored thumbnails. It needs a proper cascade.